drivers/soundwire/qcom.c

```diff
@@ -821,6 +820,9 @@ static int qcom_swrm_probe(struct platform_device *pdev)
 		goto err_clk;
 	}
 
+	ctrl->bus.md.dev.parent = dev;
+	ctrl->bus.md.dev.of_node = dev->of_node;
+	ctrl->bus.md.dev.fwnode = dev->fwnode;
```

Comment:
I think it's be simpler to add a struct device *parent argument to sdw_add_bus_master() and do the initializations in sdw_add_bus_master. This is common code that everyone needs to do, so let's factor it